app/src/controllers/user/sagas.js

```diff
@@ -0,0 +1,87 @@
+import { takeLatest, call, all, put, select } from 'redux-saga/effects';
+import { fetch } from 'common/utils/fetch';
+import { URLS } from 'common/urls';
+import { showNotification, NOTIFICATION_TYPES } from 'controllers/notification';
+import { PROJECT_MANAGER } from 'common/constants/projectRoles';
+import { userIdSelector } from './selectors';
+import { ASSIGN_TO_RROJECT, UNASSIGN_FROM_PROJECT } from './constants';
+import { assignToProjectSuccessAction, unassignFromProjectSuccessAction } from './actionCreators';
+
+function* assignToProject({ payload: project }) {
+  const userId = yield select(userIdSelector);
+  const userRole = PROJECT_MANAGER;
+  const data = {
+    userNames: {
+      [userId]: userRole,
+    },
+  };
+  try {
+    yield call(fetch, URLS.userInviteInternal(project.projectName), {
+      method: 'put',
+      data,
+    });
+    yield put(
+      assignToProjectSuccessAction({
+        projectName: project.projectName,
+        projectRole: userRole,
+        entryType: project.entryType,
+      }),
+    );
+    yield put(
+      showNotification({
+        messageId: 'assignSuccess',
+        type: NOTIFICATION_TYPES.SUCCESS,
+      }),
+    );
+  } catch (err) {
+    const error = (err.response && err.response.data && err.response.data.message) || err.message;
```

Comment:
See the https://github.com/reportportal/service-ui/pull/1310#discussion_r270774479